include/flatbuffers/base.h

```diff
@@ -236,10 +240,8 @@ namespace flatbuffers {
 
 #ifndef FLATBUFFERS_LOCALE_INDEPENDENT
   // Enable locale independent functions {strtof_l, strtod_l,strtoll_l, strtoull_l}.
-  // They are part of the POSIX-2008 but not part of the C/C++ standard.
-  // GCC/Clang have definition (_XOPEN_SOURCE>=700) if POSIX-2008.
   #if ((defined(_MSC_VER) && _MSC_VER >= 1800)            || \
-       (defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE>=700)))
+       (defined(__ANDROID_API__) && (__ANDROID_API__>=21)))
```

Comment:
What should to do if someone has `(defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE>=700))` but doesn't has ANDROID_API?
If you want to enable this for ANDROID_API>=21 you need to add the new condition with logical-OR.
Also, you can use logical-AND If you need to disable for ANDROID_API<21.

Reply:
Thanks for your comments! 

If someone enables _XOPEN_SOURCE is does not guarantee in any way that these features are available (strtoll_l etc.). It is only a request to make these features available.

You should turn this around actually:
Before any includes that use strtoll_l, if LOCALE_INDEPENDENT is 1 then _XOPEN_SOURCE should be enabled to ensure that the compiler enables these features.

Reply:
Thank you for the explanation.
There are a lot of building systems for C++: make, make, meson, scons, bazel, ...
Probably it was the wrong idea to add autodetection of the locale-independent functions both to CMake and base.h. At least for non-windows platforms.

Reply:
I think detecting if certain functions are available should be done in the build system. I think it is hard (or impossible) to do with the preprocessor. 

I think for flatbuffers there are two supported build systems cmake and gradle for android? Or is it the intent to support others as well?

Reply:
The Flatbuffers doesn't depend on build system.
I agree with you that it is hard to do with the preprocessor. This was my mistake.

Support of locale-independent "string-to-number" functions is useful for locales where decimal separator is comma ',' instead of dot '.' (https://en.wikipedia.org/wiki/Decimal_separator).
It is extremely useful if Flatbuffers used as part of a shared library that can be loaded into  process with an unknown C-locale.
I think that it would be better to remove _XOPEN_SOURCE-based detection from `base.h`.
But this change can break user code if this detection is already in use in non-windows project.

Can we use `&& !(defined(__ANDROID_API__))` as workaround?

Reply:
Very fair point, I'll have a try I think that should be ok!

Reply:
Hmmm I cannot move base.h to the top of util.cpp as the feature macros need to come first. If these come first this means that locale_independent will always be defined. 

What we could do is switch to xopen_version macro which does indicate if xopen is available from unitstd, ref https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/unistd.h.html.  This should be defined if _XOPEN_SOURCE is defined and the platform supports it. It only might break on platforms that partiallly support xopen and things like that. What do you think? And I'm a bit worried about the unistd include. Should be on any unix platform but I'm not sure.

Reply:
Could you have another look?

Reply:
The сondition check looks correct.
I'm not sure about the warning. It will take attention even if source code doesn't depend on `FLATBUFFERS_LOCALE_INDEPENDENT`.
@aardappel could you look at this PR?

Reply:
Fair enough I'll remove it!